### PR TITLE
Feature/management service

### PR DESCRIPTION
**Pull request by newtronahmed**
This pull request includes a variety of changes aimed at improving enhancing logging capabilities, updating dependencies, and refining API responses. The most important changes are summarized below:

### Dependency Management:
* Added Spring Cloud dependency management to `management/pom.xml` and included dependencies for `spring-boot-starter-actuator` and `spring-cloud-starter-netflix-eureka-client`.

### Logging Enhancements:
* Introduced `RequestLoggingInterceptor` to log incoming HTTP requests, headers, and request bodies. Integrated this interceptor into the Spring Web configuration.

### API Response Improvements:
* Updated `ProviderServiceController`, `ServiceExperienceController`, and `ServicesController` to use `ApiSuccessResponse` instead of `ApiResponse`, and added detailed Swagger annotations for better API documentation

These changes collectively enhance the application's configuration flexibility, dependency management, logging capabilities, and API response consistency.

### CODE REVIEW

File: management/src/main/java/team/proximity/management/controllers/ProviderServiceController.java [hunk 1]
```diff
@@ -28,104 +33,109 @@ public class ProviderServiceController {
...
+    @ApiResponses(value = {
+            @io.swagger.v3.oas.annotations.responses.ApiResponse(responseCode = "204", description = "Successfully deleted service experience"),
+            @ApiResponse(responseCode = "201", description = "Successfully created or updated provider service",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = ProviderService.class))),
+            @ApiResponse(responseCode = "400", description = "Invalid input"),
+            @ApiResponse(responseCode = "401", description = "You are not authorized to view the resource"),
+            @ApiResponse(responseCode = "403", description = "Accessing the resource you were trying to reach is forbidden"),
+            @ApiResponse(responseCode = "404", description = "The resource you were trying to reach is not found")
+    })
+    public ResponseEntity<ApiSuccessResponse<ProviderService>> createOrUpdateProviderService(@Validated @ModelAttribute ProviderServiceRequest providerServiceRequest) throws JsonProcessingException {
```
Comment (Zena4L) [inline management/src/main/java/team/proximity/management/controllers/ProviderServiceController.java]: Handle `JsonProcessingException` inside your service.

File: management/src/main/java/team/proximity/management/controllers/ProviderServiceController.java [hunk 2]
```diff
@@ -28,104 +33,109 @@ public class ProviderServiceController {
...
+        log.info("Processing providerService request: {}", providerServiceRequest);
 
+        ProviderService providerService = providerServiceService.createOrUpdateProviderService(providerServiceRequest);
 
-
-    @PutMapping("/{id}")
-    @Operation(summary = "Update a providerService", description = "Update a providerService by id")
-
-    public ResponseEntity<ApiResponse<ProviderService>> updateProviderService(@PathVariable UUID id, @RequestBody ProviderServiceRequest providerService) {
```
Comment (Zena4L) [inline management/src/main/java/team/proximity/management/controllers/ProviderServiceController.java:37]: Move all logics to the service. clean up the controller. 
Single Responsibility Principle

File: management/src/main/java/team/proximity/management/controllers/ProviderServiceController.java [hunk 3]
```diff
@@ -28,104 +33,109 @@ public class ProviderServiceController {
...
-    public ResponseEntity<ApiResponse<ProviderService>> getProviderServiceById(@PathVariable UUID id) {
+    @Operation(summary = "Retrieve a provider service by id", description = "Returns a provider service by id")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successfully retrieved provider service by id",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = ProviderService.class))),
+            @ApiResponse(responseCode = "401", description = "You are not authorized to view the resource"),
+            @ApiResponse(responseCode = "403", description = "Accessing the resource you were trying to reach is forbidden"),
+            @ApiResponse(responseCode = "404", description = "The resource you were trying to reach is not found")
+    })
+    public ResponseEntity<ApiSuccessResponse<ProviderService>> getProviderServiceById(@PathVariable UUID id) {
```
Comment (Zena4L) [inline management/src/main/java/team/proximity/management/controllers/ProviderServiceController.java:69]: Move all logics to the service. clean up the controller.
Single Responsibility Principle

File: management/src/main/java/team/proximity/management/validators/BookingDayHoursValidator.java [hunk 4]
```diff
@@ -11,42 +11,43 @@
 
 @Component
 public class BookingDayHoursValidator {
```
Comment (Zena4L) [inline management/src/main/java/team/proximity/management/validators/BookingDayHoursValidator.java]: Make the constructor private and also I suggest u make the class final

File: management/src/main/java/team/proximity/management/validators/upload/FileValidationContext.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    public FileValidationContext(FileValidationStrategy strategy) {
+        this.strategy = strategy;
+    }
+
+    public void setValidationStrategy(FileValidationStrategy strategy) {
+        this.strategy = strategy;
+    }
+
+    public void validate(MultipartFile file) throws InvalidFileTypeException {
+        if (strategy == null) {
```
Comment (Zena4L) [inline management/src/main/java/team/proximity/management/validators/upload/FileValidationContext.java]: Don't compare objects using "=="
Use the equals

**Comment by newtronahmed** [comment]
@Zena4L Kindly check the new changes
 Fixed the JSON processing exception handling as per the feedback.
Made BookingDayHoursValidator final and utilized a private constructor
Used object.requirenonnull for null checking

**Review by Zena4L — APPROVED** [review]

**Pull request merged**